src/backend/joanie/core/models/contracts.py

```diff
@@ -0,0 +1,232 @@
+import logging
+import textwrap
+
+from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.db import models
+from django.utils.functional import lazy
+from django.utils.translation import gettext_lazy as _
+
+import markdown
+
+from joanie.core import enums
+from joanie.core.utils import image_to_base64
+from joanie.core.utils.issuers import generate_document
+
+from .base import BaseModel
+from .products import Order
+
+logger = logging.getLogger(__name__)
+
+
+class ContractDefinition(BaseModel):
+    """
+    Contract definition describes template and markdown to generate user contracts
+    """
+
+    title = models.CharField(_("title"), max_length=255)
+    description = models.TextField(_("description"), blank=True)
+    body = models.TextField(_("body"), blank=True)
+    language = models.CharField(
+        max_length=10,
+        choices=lazy(lambda: settings.LANGUAGES, tuple)(),
+        verbose_name=_("language"),
+        help_text=_("Language of the contract definition"),
+    )
+
+    template = models.CharField(
+        _("template to generate pdf contract"),
+        max_length=255,
+        blank=True,
+        null=True,
+    )
+
+    class Meta:
+        db_table = "joanie_contract_definition"
+        verbose_name = _("Contract definition")
+        verbose_name_plural = _("Contract definitions")
+
+    def __str__(self):
+        return self.title
+
+    def get_body_in_html(self):
+        """Return the body in html format."""
+        if not self.body:
+            return ""
+        return markdown.markdown(textwrap.dedent(self.body))
+
+    def get_document_context(self, order):
+        """
+        Build the contract document context for the given language.
+        """
+        organization = order.organization
+        owner = order.owner
+        product = order.product
+
+        return {
+            "contract": {
+                "body": self.get_body_in_html(),
+                "title": self.title,
+            },
+            "course": {
+                "name": product.safe_translation_getter(
+                    "title", language_code=self.language
+                ),
+            },
+            "student": {
+                "name": owner.get_full_name() or owner.username,
+                "address": owner.addresses.filter(is_main=True).first(),
+            },
+            "organization": {
+                "logo": image_to_base64(organization.logo),
+                "name": organization.safe_translation_getter(
+                    "title", language_code=self.language
+                ),
+                "signature": image_to_base64(organization.signature),
+            },
+        }
+
+    def generate_document(self, order):
+        """
+        Generate the contract definition.
+        """
+        context = self.get_document_context(order)
+        file_bytes = generate_document(
+            path=enums.ISSUER_TEMPLATE_CONTRACT_DEFINITION, context=context
```

Comment:
The path should come from the order's product:
```suggestion
            path=order.product.contract_definition.template, context=context
```